flaky test

**Describe the bug**
there's a flaky test:

```
Test Case '-[APNSwiftJWTTests.JWTTests testJWTSigning]' started.
/private/tmp/.nio-release-tools_nHItWh/APNSwift/Tests/APNSwiftJWTTests/JWTTests.swift:77: error: -[APNSwiftJWTTests.JWTTests testJWTSigning] : XCTAssertEqual failed: ("63") is not equal to ("64")
Test Case '-[APNSwiftJWTTests.JWTTests testJWTSigning]' failed (0.288 seconds).
```

[log_swift_wmu81k.log](https://github.com/kylebrowning/APNSwift/files/3521659/log_swift_wmu81k.log)

**To Reproduce**

run `swift test --sanitize=address -c release -Xswiftc -enable-testing --disable-index-store` many times

**Expected behavior**

tests pass

**Platform:**
macOS 10.15 beta 6ish

@kylebrowning it took about 5 mins running the following command until it hit the failure for me:

```
while swift test --sanitize=address -c release -Xswiftc -enable-testing --disable-index-store; do date; done
```

Just run it, and use your computer normally for other things and check back every 5 mins if it failed, I'm sure it will at some point.


```
Test Suite 'APNSwiftRequestTests' passed at 2019-09-22 9:36:08.522 pm.
	 Executed 7 tests, with 0 failures (0 unexpected) in 0.004 (0.004) seconds
Test Suite 'JWTTests' started at 2019-09-22 9:36:08.522 pm
Test Case '-[APNSwiftJWTTests.JWTTests testJWTEncoding]' started.
Test Case '-[APNSwiftJWTTests.JWTTests testJWTEncoding]' passed (0.001 seconds).
Test Case '-[APNSwiftJWTTests.JWTTests testJWTSigning]' started.
/Users/johannes/extsrc/APNSwift/Tests/APNSwiftJWTTests/JWTTests.swift:77: error: -[APNSwiftJWTTests.JWTTests testJWTSigning] : XCTAssertEqual failed: ("63") is not equal to ("64")
Test Case '-[APNSwiftJWTTests.JWTTests testJWTSigning]' failed (0.047 seconds).
Test Suite 'JWTTests' failed at 2019-09-22 9:36:08.570 pm.
	 Executed 2 tests, with 1 failure (0 unexpected) in 0.048 (0.048) seconds
Test Suite 'apnswiftPackageTests.xctest' failed at 2019-09-22 9:36:08.570 pm.
	 Executed 12 tests, with 1 failure (0 unexpected) in 0.118 (0.119) seconds
Test Suite 'All tests' failed at 2019-09-22 9:36:08.570 pm.
	 Executed 12 tests, with 1 failure (0 unexpected) in 0.118 (0.120) seconds
```

--- a/Sources/APNSwift/APNSwiftSigner.swift
+++ b/Sources/APNSwift/APNSwiftSigner.swift
@@ -70,8 +70,9 @@ public struct APNSwiftSigner {
         let lens = Int(BN_bn2bin(.make(optional: s), &sb))
 
         var signatureBytes = ByteBufferAllocator().buffer(capacity: lenr + lens)
-        signatureBytes.writeBytes(rb[0..<lenr])
-        signatureBytes.writeBytes(sb[0..<lens])
+        let allZeroes = Array(repeating: UInt8(0), count: 32)
+        signatureBytes.writeBytes([allZeroes, rb].joined().suffix(32))
+        signatureBytes.writeBytes([allZeroes, sb].joined().suffix(32))
         return signatureBytes
     }
 }

